framework/src/constraints/AutomaticMortarGeneration.C

```diff
@@ -1739,7 +1740,10 @@ AutomaticMortarGeneration::projectSecondaryNodesSinglePair(
             // on the interface start off being perfectly aligned. In this situation, we need to
             // associate the secondary node with two different elements (and two corresponding
             // xi^(2) values.
-            if (std::abs(std::abs(xi2) - 1.) < _xi_tolerance)
+
+            // Increasing the tolerance 5 times in this if statement to avoid missing nodes due to
+            // the mismatch between sizes of primary and secondary surfaces. Still, it can happen.
+            if (std::abs(std::abs(xi2) - 1.) < _xi_tolerance * 5.0)
```

Comment:
should we just make `_xi_tolerance` 5 times bigger?

Reply:
Hmm I don't think that's the issue. We are projecting on one side first and the other side second. If we make the tolerance bigger and remove the (5) factor we are going to continue to miss the second projection and fall into that message. What makes this modification to not fall in the exception is that we are projecting on one side more `xi` than in the other. There should be a better way of doing this by using actual distances and not parametric coordinates. But I believe making the tolerance uniformly larger or smaller won't do the trick here.

Reply:
Ok. I would put that logic in the comment because it helps me understand